src/fsharp/service/ReactorListener.fs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.SourceCodeServices
+
+open System
+open System.Diagnostics
+
+type public IReactorListener =
+    abstract OnReactorPauseBeforeBackgroundWork : pauseMillis: int -> unit
+    abstract OnReactorOperationStart : userOpName: string -> opName: string -> opArg: string -> approxQueueLength: int -> unit
+    abstract OnReactorOperationEnd : userOpName: string -> opName: string -> elapsed: TimeSpan -> unit
+    abstract OnReactorBackgroundStart: bgUserOpName: string -> bgOpName: string -> bgOpArg: string -> unit
+    abstract OnReactorBackgroundCancelled : bgUserOpName: string -> bgOpName: string -> unit
+    abstract OnReactorBackgroundEnd : bgUserOpName: string -> bgOpName: string -> elapsed: TimeSpan -> unit
+
+    abstract OnSetBackgroundOp : approxQueueLength: int -> unit
+    abstract OnCancelBackgroundOp : unit -> unit
+    abstract OnEnqueueOp : userOpName: string -> opName: string -> opArg: string -> approxQueueLength: int -> unit
+
+type public EmptyReactorListener() =
+    interface IReactorListener with
+        override __.OnReactorPauseBeforeBackgroundWork _ = ()
```

Comment:
Since this is new code, can we consider using the single-underscore self identifier here?

Reply:
@cartermp, I think fcs may use a bootstrap compiler thats pre- _.SomeMethod.  I sort of half remember some issue due to fcs.